File: docs/api/agents/create_agent.py
"""Create a new agent via the Kortecx API."""
import requests, json, sys

FRONTEND = "http://localhost:3000"

agent = {
    "name": sys.argv[1] if len(sys.argv) > 1 else "Sample Agent",
    "role": sys.argv[2] if len(sys.argv) > 2 else "coder",
    "description": "A versatile AI agent for code generation and review.",
    "systemPrompt": "You are an expert software engineer. Generate clean, production-ready code with proper error handling.",
    "modelSource": "local",
    "localModelConfig": {"engine": "ollama", "modelName": "llama3.2:3b"},
    "temperature": 0.7,
    "maxTokens": 4096,
    "tags": ["python", "code-gen"],
    "capabilities": ["code-generation", "review"],
    "specializations": ["Python", "FastAPI"],
    "category": "custom",
    "complexityLevel": 3,
}

resp = requests.post(f"{FRONTEND}/api/experts", json=agent)
data = resp.json()
if resp.ok:
    e = data.get("expert", {})
    print(f"✓ Agent created: {e.get('id')}")
    print(f"  Name: {e.get('name')}, Role: {e.get('role')}")
else:
    print(f"✗ Failed: {data.get('error', resp.text)}")

File: docs/api/typescript/create_agent.ts
/**
 * Create an agent via the Kortecx API (TypeScript)
 * Run: npx tsx docs/api/typescript/create_agent.ts
 */
const FRONTEND = "http://localhost:3000";

async function createAgent() {
  const agent = {
    name: process.argv[2] || "TS Sample Agent",
    role: "coder",
    description: "TypeScript-based agent for code generation.",
    systemPrompt: "You are an expert TypeScript developer.",
    modelSource: "local",
    localModelConfig: { engine: "ollama", modelName: "llama3.2:3b" },
    temperature: 0.7,
    maxTokens: 4096,
    tags: ["typescript", "code-gen"],
    capabilities: ["code-generation"],
    category: "custom",
    complexityLevel: 3,
  };

  const resp = await fetch(`${FRONTEND}/api/experts`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(agent),
  });
  const data = await resp.json();
  const e = data.expert || {};
  console.log(resp.ok ? `✓ Agent: ${e.id}` : `✗ Failed: ${data.error}`);
}

createAgent().catch(console.error);
